### ISSUE TITLE
Plugin no longer works as of OBS 28.0.0-rc1 (Mac)

### ISSUE THREAD

**Issue by josephjwilk** [phandasm/waveform#26]
The plugin worked with 28.0 Beta 2 (running MacOS 12.5.1 on an M1 Pro) but as of upgrading to OBS 28.0.0-rc1 is now unable to be added as a source. Existing instances are broken (in that there is no reaction, and the source can't have its properties adjusted).

Thanks for a great plugin, by the way. I've really enjoyed using it.

**Comment by fzwoch** [comment]
Check this commit. Plugins for ARM64 on macOS need to be wrapped in the `.plugin` format. Legacy `.so` files and locations not longer work.

[b543f96a3b294a338063c3fec7759c51a75560de](https://github.com/obsproject/obs-studio/commit/b543f96a3b294a338063c3fec7759c51a75560de)

**Comment by phandasm** [comment]
Thanks for the heads up, I'll try to get a compatible build out soon but I don't have a mac on hand to test with at the moment so we'll see how that goes.

**Comment by fzwoch** [comment]
I believe a minimal bundle does work. Like this:

```bash
$ find obs-teleport.plugin/
obs-teleport.plugin/
obs-teleport.plugin/Contents
obs-teleport.plugin/Contents/Info.plist
obs-teleport.plugin/Contents/MacOS
obs-teleport.plugin/Contents/MacOS/obs-teleport
```
```bash
$ cat obs-teleport.plugin/Contents/Info.plist 
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>CFBundleExecutable</key>
	<string>obs-teleport</string>
</dict>
```

**Comment by daveyg71** [comment]
I'm down to test it for you, if needed.

**Comment by phandasm** [comment]
So I still have unresolved build system issues targeting OBS 28 but in the interest of not delaying this any further I've attempted a workaround: https://github.com/phandasm/waveform/releases/tag/v1.5.0-beta1

If anyone would like to test it please do let me know if it works.

**Comment by kilinbox** [comment]
thanks , v1.5.0-beta1
It worked on my macOS OBS 28.0 (arm64).

**Comment by daveyg71** [comment]
I cant get it to work on my M1 Mac mini

**Comment by fzwoch** [comment]
I tried too. macOS was just complaining, that it is broken and asks whether I want to move it to the trash bin.

**Comment by josephjwilk** [comment]
I can confirm that the v1.5.0-beta1 also does not work on my M1 Pro system. I get the same message as as @fzwoch when opening OBS.

**Comment by phandasm** [comment]
Well 25% success rate isn't exactly reassuring, I don't suppose this could be a security settings issue again?

**Comment by fzwoch** [comment]
Usually, when not signed by a developer certificate etc macOS just informs about it and one have to acknowledge it in security settings to run it anyway. The "move to trash" message seems a bit more forced.

**Comment by phandasm** [comment]
Odd, maybe it's one of those things where it works for people who have xcode installed or something, must be missing some compiler flags.

**Comment by kilinbox** [comment]
Oops, I did not know that this does not work in some environments.
Here is a video of my mac mini in action.

https://user-images.githubusercontent.com/17141779/188266144-28e38674-65b1-4608-a8a5-c03092823442.mp4

**Comment by fzwoch** [comment]
The plugin is indeed fine. It is yet another macOS annoyance.

When downloaded with Safari the file gets quarantined and cannot be loaded.

One has to manually remove the quarantine flag:

```
xattr -d com.apple.quarantine ~/Library/Application\ Support/obs-studio/plugins/waveform.plugin
```

Alternatively, download via wget or probably any other browser.

**Comment by josephjwilk** [comment]
I can confirm that @fzwoch's terminal solution worked for me, as the file was also being quarantine flagged via Chrome.

**Comment by oparada1988** [comment]
Came to confirm that v1.5.0-beta1 works on Ubuntu 22.04.1 LTS, OBS V28

**Comment by ShKlinkenberg** [comment]
Also can confirm v1.5.0-beta1 on macOS m1 is working after removing the quarantine

**Comment by phandasm** [comment]
So I switched to actual packages in an attempt to mitigate the quarantine issue: https://github.com/phandasm/waveform/releases/tag/v1.5.0-rc1

I tested the x86 version but it'd be nice to have confirmation that the ARM version works too, if anyone wants to try it.

**Comment by josephjwilk** [comment]
The plugin works with no issues or command line requirements; however, the package is installing the plugin in the system library, rather than the user library. My installation of OBS is unable to see plugins in that library folder.

**Comment by phandasm** [comment]
I swapped the arm64 package on the release page: https://github.com/phandasm/waveform/releases/tag/v1.5.0-rc1
I'm fairly certain it should be fine now, but do let me know if it's not.
I'll be pushing a full release pretty soon otherwise.

**Comment by josephjwilk** [comment]
I can confirm that this package worked for me!

**Comment by phandasm** [comment]
Fixed as of 1.5.0
